Core/Node-API/CMakeLists.txt

```
@@ -95,6 +95,49 @@ if(NAPI_BUILD_ABI)
 
             set(LINK_LIBRARIES ${LINK_LIBRARIES}
                 PUBLIC "${V8_ANDROID_DIR}/jni/${ANDROID_ABI}/libv8android.so")
+        elseif(WIN32)
+            set_cpu_platform_arch()
+            set(V8_VERSION "8.4.371.15")
+            #install_v8_nuget(${V8_VERSION} ${CPU_ARCH})
+            download_nuget()
+            set(V8_PACKAGE_PATH "${NUGET_PATH}/packages/v8-v142-${CPU_ARCH}.${V8_VERSION}")
+            set(V8_PACKAGE_PATH "${NUGET_PATH}/packages/v8-v142-${CPU_ARCH}.${V8_VERSION}")
+            set(V8_REDIST_PACKAGE_PATH "${NUGET_PATH}/packages/v8.redist-v142-${CPU_ARCH}.${V8_VERSION}")
+
+            add_library(v8_libbase SHARED IMPORTED)
+            set_target_properties(v8_libbase PROPERTIES IMPORTED_IMPLIB "${V8_PACKAGE_PATH}/lib/Release/v8_libbase.dll.lib")
+            add_library(v8_libplatform SHARED IMPORTED)
+            set_target_properties(v8_libplatform PROPERTIES IMPORTED_IMPLIB "${V8_PACKAGE_PATH}/lib/Release/v8_libplatform.dll.lib")
+            add_library(v8 SHARED IMPORTED)
+            set_target_properties(v8 PROPERTIES IMPORTED_IMPLIB "${V8_PACKAGE_PATH}/lib/Release/v8.dll.lib")
+            target_link_libraries(v8 INTERFACE v8_libbase INTERFACE v8_libplatform)
+            target_include_directories(v8 INTERFACE "${V8_PACKAGE_PATH}/include")
+            
+            set(V8_DIST
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/icudtl.dat"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/icui18n.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/icuuc.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/v8.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/v8_libbase.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/v8_libplatform.dll"
+                "${V8_REDIST_PACKAGE_PATH}/lib/Release/zlib.dll")
+
+            # only 1 imported location per library -> Adding 1 library per file
+            foreach(V8FILE ${V8_DIST} )
+                get_filename_component(V8FILE_NAME "${V8FILE}" NAME_WE)
+                add_library("v8::${V8FILE_NAME}" SHARED IMPORTED)
+                set_target_properties("v8::${V8FILE_NAME}" PROPERTIES IMPORTED_IMPLIB "${V8_PACKAGE_PATH}/lib/Release/v8_libbase.dll.lib")
+                set_target_properties("v8::${V8FILE_NAME}" PROPERTIES IMPORTED_LOCATION ${V8FILE})
+            endforeach()
+
+            set(LINK_LIBRARIES v8 v8::icudtl v8::icui18n v8::icuuc v8::v8 v8::v8_libbase v8::v8_libplatform v8::zlib)
```

Should probably follow the pattern of appending to existing and indicate PRIVATE, INTERFACE, PUBLIC.
```suggestion
            set(LINK_LIBRARIES ${LINK_LIBRARIES}
                PRIVATE v8::icudtl
                PRIVATE v8::icui18n
                PRIVATE v8::icuuc
                PRIVATE v8::v8
                PRIVATE v8::v8_libbase
                PRIVATE v8::v8_libplatform
                PRIVATE v8::zlib)
```

---

Fixed with v8 added as public